### DISCUSSION TITLE
Using filters on opinions api endpoint

### DISCUSSION THREAD

**Discussion by jumpingtom1** [freelawproject/courtlistener#3959] [Q&A] [answered]
Hi -
I'm getting started with a project using the Court Listener API. It's a great resource and I'm excited. But, I'm having trouble using the filters with the opinions endpoint. Say I want to download text of opinions from Oregon Supreme Court and Oregon Court of Appeals containing the term "dog".

First, I'm calling the search endpoint with this URL/query:
`REQUEST_URL = f'https://www.courtlistener.com/api/rest/v3/search/?type=o&q="dog"&court=or orctapp'`

The response shows that there are 718 cases. I will then run through the results and collect the "id" field. The first id is 1163781, State v. Krummacher

Next, I call the opinion endpoint using the case id:
`REQUEST_URL = f'https://www.courtlistener.com/api/rest/v3/opinions/?id__exact=1163781'`

But, the result here isn't filtered by the id. Instead, appears to just show results from opinion endpoint on whatever happens to be there first. I've tried a variety of different filters with the opinions endpoint and none seem to filter the results. What am I doing wrong?

Thanks so much!

**Comment by mlissner** [comment, marked as answer]
Ha, well, you're overthinking it, I fear. Do this:

https://www.courtlistener.com/api/rest/v3/opinions/1163781/

**Reply by jumpingtom1** [reply]
Awesome. Thanks.
